src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/DynamicallyAccessedMemberTypes.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    /// <summary>
+    /// Specifies the types of members that are dynamically accessed.
+    ///
+    /// This enumeration has a <see cref="FlagsAttribute"/> attribute that allows a
+    /// bitwise combination of its member values.
+    /// </summary>
+    [Flags]
+    public enum DynamicallyAccessedMemberTypes
+    {
+        /// <summary>
+        /// Specifies no members.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        /// Specifies the default, parameterless public constructor.
+        /// </summary>
+        DefaultConstructor = 0x0001,
```

Comment:
We separate this one out specifically because it's commonly needed for things like the `new` constraint / `Activator.CreateInstance`?

Reply:
Yes - it feels like this is a common enough scenario that it's worth having a separate entry for it that doesn't drag in the rest of the constructors.

Reply:
What happens if I use 0x0002... that means all public ctors except the default?

Reply:
We should probably treat all values that cannot be produced by `or`ing the defined flags as errors.

Reply:
Instead of DefaultCtor, Should we instead have a string-based overload that lets you specify a member name? Presumably there will be other cases that helps, such as if the implementation uses a specific member of the type, like Count?

Reply:
Would we have use cases for the string based overload in the framework? It feels like if we are annotating a specific System.Type instance with a specific member, we might as well just grab that specific MemberInfo at the time things are statically known and then we don't need to annotate at all because instead of flowing the annotated System.Type, we would flow the `MemberInfo` directly.

Default constructor is special because the type is typically passed to Activator that doesn't deal with ConstructorInfos (and switching to ConstructorInfo would be a perf deoptimization).

Reply:
> Would we have use cases for the string based overload in the framework?

I don't know yet.  It just jumps out at me any time we special-case something like this in public surface area.  Then again, this particular ctor is already special-cased by the C# language in numerous ways, so, maybe the inconsistency is the greater good.

Reply:
> Would we have use cases for the string based overload in the framework?

One place that I recently saw is this:

https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/MaxLengthAttribute.cs#L124-L129

However, we don't have a `Type` to annotate - it is called taking an `object` from `MaxLengthAttribute.IsValid`. So I don't think there is a way the linker could know all the types that are passed into that method.

Reply:
Currently whenever there's a call to `Object.GetType()` it's game over for the linker. There might be some magic we could do in "some" cases, but in general it won't work. So the above sample is already very problematic from the linker's point of view.